Sources/Core/JavaModelRenderer.swift

```diff
@@ -74,11 +76,36 @@ public struct JavaModelRenderer: JavaFileRenderer {
         return [props, bitmasks, [bits]]
     }
 
-    func renderModelGetters(modifiers: JavaModifier = [.public]) -> [JavaIR.Method] {
-        let getters = transitiveProperties.map { param, schemaObj in
-            JavaIR.method(modifiers, self.typeFromSchema(param, schemaObj) + " get" + param.snakeCaseToCapitalizedPropertyName() + "()") { [
+    func propertyGetterForParam(param: String, schemaObj: SchemaObjectProperty) -> JavaIR.Method {
+        // For Booleans, Integers and Doubles, make the getter method @NonNull and squash to a default value if necessary.
+        // This makes callers less susceptible to null pointer exceptions.
+        switch schemaObj.schema {
+        case .boolean:
```

Comment:
Do we need to add integer enums here?
I'd encourage you to add an extension to `Schema` to note what types are primitives for Java. We have this for ObjC
```
extension Schema {
    var isObjCPrimitiveType: Bool {
        switch self {
        case .boolean, .integer, .enumT, .float:
            return true
        default:
            return false
        }
    }
```
I think this method is just trying to state `@NonNull` should be used whenever the type is a primitive type. The alternative is updating `typeFromSchema` to return `@NonNull` if the type is a primitive type which would modify any reference the properties type

Reply:
> Do we need to add integer enums here?

Enums could be a bit more dangerous. I don't know. What do you think? I'm inclined to leave those nullable.

> I'd encourage you to add an extension to Schema to note what types are primitives for Java. We have this for ObjC. I think this method is just trying to state @NonNull should be used whenever the type is a primitive type. 

In addition to switching to `@NonNull` The body of the getter method does change slightly between the primitive types, so it can't be generalized. I'm not sure we gain much from having a `isJavaPrimitiveType` method?

Reply:
Thats fine, just more of a question TBH. If it isn't a useful addition then there isn't a point in adding it.

I think the `enum` suggestion was before I remembered that these are actually reference types and full-blown objects which  differs from c languages which are usually a type-erased alias for a primitive type (int).